### DISCUSSION TITLE
Is it possible to make snapshots with different languages?

### DISCUSSION THREAD

**Discussion by haemi** [pointfreeco/swift-snapshot-testing#513] [Q&A]
It would be great to make different snapshots for device sizes, ContentSizeCategories, interface styles - and languages/localizations. I achieved everything except the latter; is this possible somehow?

**Comment by stephencelis** [comment]
Hi @haemi, thanks for bringing this up! I'm going to convert this to a discussion, though, since we try to keep the issues section more focussed on bugs in the library.

It's definitely possible to screen shot different languages and localizations in a single test run, but generally requires you to control these dependencies in your application, and I'm not sure it's feasible or possible for SnapshotTesting to facilitate this work.

As an example, when we were at Kickstarter we had a global, controllable "environment" that could have its language overridden, and then in tests we could loop over all the supported languages and snapshot them. Here's an example from the open source repo:

https://github.com/kickstarter/ios-oss/blob/1c9b83e4fef801c05ebd2b989a02fb5124fb4376/Kickstarter-iOS/Views/Controllers/LoginViewControllerTests.swift#L12-L22

So as long as your application provides a means for overriding the language/locale, you should be able to snapshot things in this way. If you're snapshotting SwiftUI views this could be as simple as tacking on an `.environment` modifier to the view being snapshot.
